apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/attach-sidebar.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import { dispatch, select, subscribe } from '@wordpress/data';
+import React, { render } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from './constants';
+import WpcomBlockEditorNavSidebar from './wpcom-block-editor-nav-sidebar';
+import './style.scss';
+
+async function findElement( selector: string, timeoutMs = 5000 ) {
+	let pendingQuery;
+	const pollForLoadedFlag = new Promise< HTMLElement >( ( resolve ) => {
+		const runQuery = () => {
+			const element = document.querySelector( selector ) as HTMLElement;
+
+			if ( ! element ) {
+				pendingQuery = setTimeout( runQuery, 200 );
+				return;
+			}
+
+			resolve( element );
+		};
+
+		runQuery();
+	} );
+
+	const timeout = new Promise< 'timeout' >( ( resolve ) =>
+		setTimeout( resolve, timeoutMs, 'timeout' )
+	);
+
+	const finishCondition = await Promise.race( [ pollForLoadedFlag, timeout ] );
+	clearTimeout( pendingQuery );
+
+	if ( finishCondition === 'timeout' ) {
+		return undefined;
+	}
+
+	return finishCondition;
+}
+
+async function attachSidebar() {
+	const editorLayoutContainer = await findElement( '.edit-post-layout' );
+	if ( ! editorLayoutContainer ) {
+		return;
+	}
+
+	const closePostButton = await findElement( '.edit-post-fullscreen-mode-close' );
+	if ( ! closePostButton ) {
+		return;
+	}
+
+	editorLayoutContainer.classList.add( 'is-wpcom-block-editor-nav-sidebar-attached' );
+
+	closePostButton.addEventListener( 'click', ( ev ) => {
+		ev.preventDefault();
+		dispatch( STORE_KEY ).toggleSidebar();
+	} );
+
+	let sidebarExpanded = false;
+	subscribe( () => {
+		const newSidebarState = select( STORE_KEY ).isSidebarOpened();
+		if ( sidebarExpanded === newSidebarState ) {
+			return;
+		}
+
+		sidebarExpanded = newSidebarState;
+
+		if ( sidebarExpanded ) {
+			editorLayoutContainer.classList.add( 'is-wpcom-block-editor-nav-sidebar-opened' );
+		} else {
+			editorLayoutContainer.classList.remove( 'is-wpcom-block-editor-nav-sidebar-opened' );
+		}
+	} );
+
+	const sidebarContainer = document.createElement( 'div' );
+	document.body.appendChild( sidebarContainer );
```

Comment:
Would it make sense to inject this with Gutenberg's `registerPlugin` function?

Reply:
That looks like what I should be using. It renders all the plugins inside a hidden div though, and I'm not sure how I'm supposed to get my UI shown because it doesn't look like there's a way to break out of that hidden div.